+ source ../common/common_lib.sh
++ source /root/mugen/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Thu Apr 27 02:03:43 2023 - INFO  - Start environmental preparation.
+ DNF_INSTALL samba
+ pkgs=samba
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen/libs/locallibs/rpm_manage.py install --pkgs samba --node 1 --tempfile ''
+ tmpfile2=/tmp/tmpplbct1oy
+ '[' -z '' ']'
+ tmpfile=/tmp/tmpplbct1oy
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Thu Apr 27 02:04:45 2023 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Thu Apr 27 02:04:46 2023 - INFO  - Start testing...
+ test_execution nmb.service
+ service=nmb.service
++ date '+%Y-%m-%d %T'
+ log_time='2023-04-27 02:04:46'
+ test_restart nmb.service
+ service=nmb.service
+ systemctl restart nmb.service
Job for nmb.service failed because the control process exited with error code.
See "systemctl status nmb.service" and "journalctl -xeu nmb.service" for details.
+ CHECK_RESULT 1 0 0 'nmb.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='nmb.service restart failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'nmb.service restart failed'
+ LOG_ERROR 'nmb.service restart failed'
+ message='nmb.service restart failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'nmb.service restart failed'
Thu Apr 27 02:04:47 2023 - ERROR - nmb.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 34'
+ message='../common/common_lib.sh line 34'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 34'
Thu Apr 27 02:04:48 2023 - ERROR - ../common/common_lib.sh line 34
+ '[' 0 -eq 1 ']'
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status nmb.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'nmb.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='nmb.service restart failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'nmb.service restart failed'
+ LOG_ERROR 'nmb.service restart failed'
+ message='nmb.service restart failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'nmb.service restart failed'
Thu Apr 27 02:04:56 2023 - ERROR - nmb.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Thu Apr 27 02:04:57 2023 - ERROR - ../common/common_lib.sh line 37
+ '[' 0 -eq 1 ']'
+ return 0
+ systemctl stop nmb.service
+ CHECK_RESULT 0 0 0 'nmb.service stop failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='nmb.service stop failed'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status nmb.service
+ grep 'Active: inactive'
+ CHECK_RESULT 1 0 0 'nmb.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='nmb.service stop failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'nmb.service stop failed'
+ LOG_ERROR 'nmb.service stop failed'
+ message='nmb.service stop failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'nmb.service stop failed'
Thu Apr 27 02:05:04 2023 - ERROR - nmb.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Thu Apr 27 02:05:05 2023 - ERROR - ../common/common_lib.sh line 42
+ '[' 0 -eq 1 ']'
+ return 0
+ systemctl start nmb.service
Job for nmb.service failed because the control process exited with error code.
See "systemctl status nmb.service" and "journalctl -xeu nmb.service" for details.
+ CHECK_RESULT 1 0 0 'nmb.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='nmb.service start failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'nmb.service start failed'
+ LOG_ERROR 'nmb.service start failed'
+ message='nmb.service start failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'nmb.service start failed'
Thu Apr 27 02:05:07 2023 - ERROR - nmb.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 44'
+ message='../common/common_lib.sh line 44'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 44'
Thu Apr 27 02:05:08 2023 - ERROR - ../common/common_lib.sh line 44
+ '[' 0 -eq 1 ']'
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ grep 'Active: active'
+ systemctl status nmb.service
+ CHECK_RESULT 1 0 0 'nmb.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='nmb.service start failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'nmb.service start failed'
+ LOG_ERROR 'nmb.service start failed'
+ message='nmb.service start failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'nmb.service start failed'
Thu Apr 27 02:05:15 2023 - ERROR - nmb.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Thu Apr 27 02:05:16 2023 - ERROR - ../common/common_lib.sh line 47
+ '[' 0 -eq 1 ']'
+ return 0
+ test_enabled nmb.service
+ service=nmb.service
++ systemctl is-enabled nmb.service
+ state=disabled
+ '[' disabled == enabled ']'
+ '[' disabled == disabled ']'
++ systemctl enable nmb.service
++ grep 'Created symlink'
++ awk '{print $3}'
+ symlink_file=/etc/systemd/system/multi-user.target.wants/nmb.service
+ find /etc/systemd/system/multi-user.target.wants/nmb.service
/etc/systemd/system/multi-user.target.wants/nmb.service
+ CHECK_RESULT 0 0 0 'nmb.service enable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='nmb.service enable failed'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl disable nmb.service
Removed /etc/systemd/system/multi-user.target.wants/nmb.service.
+ find /etc/systemd/system/multi-user.target.wants/nmb.service
find: ‘/etc/systemd/system/multi-user.target.wants/nmb.service’: No such file or directory
+ CHECK_RESULT 1 0 1 'nmb.service disable failed'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='nmb.service disable failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ journalctl --since '2023-04-27 02:04:46' -u nmb.service
+ grep -i 'fail\|error'
+ grep -v -i 'DEBUG\|INFO\|WARNING'
Apr 27 02:04:46 openeuler-riscv64 nmbd[2002]: /usr/sbin/nmbd: error while loading shared libraries: libmessages-util-samba4.so: cannot open shared object file: No such file or directory
Apr 27 02:04:46 openeuler-riscv64 systemd[1]: nmb.service: Failed with result 'exit-code'.
Apr 27 02:04:46 openeuler-riscv64 systemd[1]: Failed to start Samba NMB Daemon.
Apr 27 02:05:06 openeuler-riscv64 nmbd[2128]: /usr/sbin/nmbd: error while loading shared libraries: libmessages-util-samba4.so: cannot open shared object file: No such file or directory
Apr 27 02:05:06 openeuler-riscv64 systemd[1]: nmb.service: Failed with result 'exit-code'.
Apr 27 02:05:06 openeuler-riscv64 systemd[1]: Failed to start Samba NMB Daemon.
+ CHECK_RESULT 0 0 1 'There is an error message for the log of nmb.service'
+ actual_result=0
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of nmb.service'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 1 -eq 0 ']'
+ test 0x == 0x
+ test -n 'There is an error message for the log of nmb.service'
+ LOG_ERROR 'There is an error message for the log of nmb.service'
+ message='There is an error message for the log of nmb.service'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'There is an error message for the log of nmb.service'
Thu Apr 27 02:05:29 2023 - ERROR - There is an error message for the log of nmb.service
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 28'
+ message='../common/common_lib.sh line 28'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 28'
Thu Apr 27 02:05:30 2023 - ERROR - ../common/common_lib.sh line 28
+ '[' 0 -eq 1 ']'
+ return 0
+ systemctl start nmb.service
Job for nmb.service failed because the control process exited with error code.
See "systemctl status nmb.service" and "journalctl -xeu nmb.service" for details.
+ sed -i 's\ExecStart=/usr/sbin/nmbd\ExecStart=/usr/sbin/nmbd -D\g' /usr/lib/systemd/system/nmb.service
+ systemctl daemon-reload
+ systemctl reload nmb.service
nmb.service is not active, cannot reload.
+ CHECK_RESULT 1 0 0 'nmb.service reload failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='nmb.service reload failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'nmb.service reload failed'
+ LOG_ERROR 'nmb.service reload failed'
+ message='nmb.service reload failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'nmb.service reload failed'
Thu Apr 27 02:05:37 2023 - ERROR - nmb.service reload failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_nmb.sh line 35'
+ message='oe_test_service_nmb.sh line 35'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_nmb.sh line 35'
Thu Apr 27 02:05:38 2023 - ERROR - oe_test_service_nmb.sh line 35
+ '[' 0 -eq 1 ']'
+ return 0
+ systemctl status nmb.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'nmb.service reload causes the service status to change'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='nmb.service reload causes the service status to change'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'nmb.service reload causes the service status to change'
+ LOG_ERROR 'nmb.service reload causes the service status to change'
+ message='nmb.service reload causes the service status to change'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'nmb.service reload causes the service status to change'
Thu Apr 27 02:05:39 2023 - ERROR - nmb.service reload causes the service status to change
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_nmb.sh line 37'
+ message='oe_test_service_nmb.sh line 37'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_nmb.sh line 37'
Thu Apr 27 02:05:40 2023 - ERROR - oe_test_service_nmb.sh line 37
+ '[' 0 -eq 1 ']'
+ return 0
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Thu Apr 27 02:05:41 2023 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 8
+ test 8 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Thu Apr 27 02:05:42 2023 - ERROR - failed to execute the case.
+ exit 8
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Thu Apr 27 02:05:43 2023 - INFO  - start environment cleanup.
+ sed -i 's\ExecStart=/usr/sbin/nmbd -D\ExecStart=/usr/sbin/nmbd\g' /usr/lib/systemd/system/nmb.service
+ systemctl daemon-reload
+ systemctl reload nmb.service
nmb.service is not active, cannot reload.
+ systemctl stop nmb.service
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmpplbct1oy ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmpplbct1oy
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Thu Apr 27 02:06:20 2023 - INFO  - Finish environment cleanup!
